src/components/Payment/PaymentForm.tsx

```diff
@@ -0,0 +1,100 @@
+import "./PaymentForm.css"
+import { useStripe, useElements, PaymentElement } from "@stripe/react-stripe-js"
+import { Layout, LayoutObject } from "@stripe/stripe-js"
+import { useContext, useEffect, useState } from "react"
+import { OnboardingContext } from "../OnboardingForm/Context"
+
+export default function PaymentForm() {
+    const stripe = useStripe()
+    const elements = useElements()
+    
+    const [paymentError, setPaymentError] = useState<string>("")
+    const [isLoading, setIsLoading] = useState<boolean>(false)
+    const { setPayment } = useContext(OnboardingContext)
+
+    useEffect(() => {
+        if (!stripe) {
+            return
+        }
+
+        const clientSecret = new URLSearchParams(window.location.search).get(
+            "payment_intent_client_secret"
+        );
+      
+        if (!clientSecret) {
+            return;
+        }
+
+        // stripe.retrievePaymentIntent(clientSecret).then(({ paymentIntent }) => {
+        //     switch (paymentIntent!.status) {
+        //       case "succeeded":
+        //         // setMessage("Payment succeeded!");
+        //         break;
+        //       case "processing":
+        //         // setMessage("Your payment is processing.");
+        //         break;
+        //       case "requires_payment_method":
+        //         // setMessage("Your payment was not successful, please try again.");
+        //         break;
+        //       default:
+        //         // setMessage("Something went wrong.");
+        //         break;
+        //     }
+        // });
+    }, [stripe])
+      
+    const paymentElementOptions: {
+        layout: Layout | LayoutObject | undefined
+    } = {
+        layout: "tabs"
+    }
+
+    const handleSubmit = async (e: { preventDefault: () => void }) => {
+        e.preventDefault();
+    
+        if (!stripe || !elements) {
+          // Stripe.js hasn't yet loaded.
+          // Make sure to disable form submission until Stripe.js has loaded.
+          return;
+        }
+    
+        setIsLoading(true);
+
+        const { error, paymentIntent } = await stripe.confirmPayment({
+          elements,
+          confirmParams: {
+            // Make sure to change this to your payment completion page
+            return_url: "http://localhost:5173/dashboard", // make a payment success page
```

Comment:
This could be an env variable where the base url can be switched to the actual url in production.

Reply:
Okay, sounds good